### PR TITLE
Parallel estimate segment size

### PR DESCRIPTION
**Pull request by xige-16**
fix issue: #12073 
Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

File: internal/querycoord/segment_allocator.go [hunk 1]
```diff
@@ -99,13 +100,24 @@ func shuffleSegmentsToQueryNodeV2(ctx context.Context, reqs []*querypb.LoadSegme
...
+	dataSizePerReq := make([]int64, len(reqs))
+	estimateError := make([]error, len(reqs))
+	var estimateWg sync.WaitGroup
+	estimateReqFn := func(offset int, req *querypb.LoadSegmentsRequest) {
+		defer estimateWg.Done()
+		dataSizePerReq[offset], estimateError[offset] = cluster.estimateSegmentsSize(req)
+	}
+	for offset, req := range reqs {
+		estimateWg.Add(1)
+		tempReq := req
```
Comment (congqixia) [inline internal/querycoord/segment_allocator.go:112]: Copying here is not necessary. This is not a closure catch.

Comment (xige-16) [reply internal/querycoord/segment_allocator.go]: fixed

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**